### PR TITLE
Expose properties of HttpProxyAgent

### PR DESCRIPTION
**Pull request by steprescott**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Increase the version number in the header if appropriate.

### CODE REVIEW

**Comment by steprescott** [comment]
The bot didn't tag the previous contributor @mrmlnc. Would be nice to get your comments.

File: types/http-proxy-agent/index.d.ts [hunk 1]
```diff
@@ -10,6 +11,11 @@ import { Url } from 'url';
 
 declare class HttpProxyAgent extends Agent {
     constructor(options: string | Url);
+
+	options: Url;
```
Comment (mrmlnc) [inline types/http-proxy-agent/index.d.ts]: I think there should be 4 spaces.

Comment (mrmlnc) [inline types/http-proxy-agent/index.d.ts]: I don't see these properties in agent implementation: `options`, `timeout`. This is the hidden/private field inside the agent (`http.Agent`)? If so, these types must be described in the [`node`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node) package.

**Review by mrmlnc — CHANGES_REQUESTED** [review]
I'm sorry, but I have a question about the fields.

File: types/http-proxy-agent/index.d.ts (hunk 1 above)
Comment (steprescott) [reply types/http-proxy-agent/index.d.ts]: Thanks for your review.

I am not sure where these properties come from. I can see that the `http-proxy-agent` declares `var proxy` and that is the only one I need. I'm happy to remove the other properties if this is what you were suggesting.

The way I found these properties was by logging the agent.

```typescript
let agent = options.agent as HttpProxyAgent
console.log(agent)
```

That give this output

```typescript
HttpProxyAgent {
  _events: {},
  _eventsCount: 0,
  _maxListeners: undefined,
  _promisifiedCallback: true,
  timeout: null,
  options:
   Url {
     protocol: 'http:',
     slashes: true,
     auth: null,
     host: 'localhost:30002',
     port: '30002',
     hostname: 'localhost',
     hash: null,
     search: null,
     query: null,
     pathname: '/',
     path: '/',
     href: 'http://localhost:30002/' },
  secureProxy: false,
  proxy:
   { protocol: 'http:',
     slashes: true,
     auth: null,
     host: 'localhost',
     port: 30002,
     hostname: 'localhost',
     hash: null,
     search: null,
     query: null,
     href: 'http://localhost:30002/' },
  callback: [Function] }
```

**Comment by steprescott** [comment]
@mrmlnc I have committed your suggested changes and added comments that should help explain why I added the `options` & `timeout` properties.

Any more issues please let me know. Thanks again for your time.

File: types/http-proxy-agent/index.d.ts (hunk 1 above)
Comment (mrmlnc) [reply types/http-proxy-agent/index.d.ts]: This is should be a part of the Node.js typings:

https://github.com/nodejs/node/blob/master/lib/_http_agent.js#L51

**Review by mrmlnc — APPROVED** [review]

**Pull request merged**